src/components/views/messages/TextualBody.js

```diff
@@ -143,9 +143,15 @@ module.exports = React.createClass({
         if (this.props.showUrlPreview && !this.state.links.length) {
             var links = this.findLinks(this.refs.content.children);
             if (links.length) {
-                this.setState({ links: links.map((link)=>{
-                    return link.getAttribute("href");
-                })});
+                // de-dup the links
+                const seen = new Set();
+                links = links.filter((link) => {
```

Comment:
Also, don't you need to be doing `links = links.map((link) => link.getAttribute("href"));` before putting them in the set?